### ISSUE TITLE
No gradients provided for any variable

### ISSUE THREAD

**Issue by ZJUzxf** [d909b/cxplain#4]
I try the CIFAR10 and MNIST example codes.
`explainer = CXPlain(explained_model, model_builder, masking_operation, loss, 
                    num_models=5, downsample_factors=downsample_factors, flatten_for_explained_model=True)
explainer.fit(x_train, y_train);`
but there are the same problem when I run the codes.
`ValueError: No gradients provided for any variable: ["<tf.Variable 'conv2d_130/kernel:0' shape=(3, 3, 1, 8) dtype=float32>", "<tf.Variable 'conv2d_130/bias:0' shape=(8,) dtype=float32>", "<tf.Variable 'conv2d_131/kernel:0' shape=(3, 3, 8, 8) dtype=float32>", "<tf.Variable 'conv2d_131/bias:0' shape=(8,) dtype=float32>"`...
Can you please tell me how to solve this problem?
Thanks

**Comment by Magolor** [comment]
> I try the CIFAR10 and MNIST example codes.
> `explainer = CXPlain(explained_model, model_builder, masking_operation, loss, num_models=5, downsample_factors=downsample_factors, flatten_for_explained_model=True) explainer.fit(x_train, y_train);`
> but there are the same problem when I run the codes.
> `ValueError: No gradients provided for any variable: ["<tf.Variable 'conv2d_130/kernel:0' shape=(3, 3, 1, 8) dtype=float32>", "<tf.Variable 'conv2d_130/bias:0' shape=(8,) dtype=float32>", "<tf.Variable 'conv2d_131/kernel:0' shape=(3, 3, 8, 8) dtype=float32>", "<tf.Variable 'conv2d_131/bias:0' shape=(8,) dtype=float32>"`...
> Can you please tell me how to solve this problem?
> Thanks

The same here, have you solved it?

**Comment by adityasaini70** [comment]
##  **TL;DR: Replace the 4th block with the following code**

```
import tensorflow as tf
tf.compat.v1.disable_v2_behavior()
explainer = CXPlain(explained_model, model_builder, masking_operation, loss, num_models=10)
explainer.fit(x_train, y_train);
```


If your Tensorflow version is >= 2, then you might face this problem. I think CXPlain was built on top of Tensorflow 1.x versions. The developers at Tensorflow made a lot of changes when they launched Tensorflow 2.0.

Still, you can work around it by _disabling_ the behavior of the 2nd version and adding these lines in the 4th block:

```
import tensorflow as tf
tf.compat.v1.disable_v2_behavior()

```
